pkg/shed/index.go

```diff
@@ -45,7 +46,41 @@ type Item struct {
 	BinID           uint64
 	PinCounter      uint64 // maintains the no of time a chunk is pinned
 	Tag             uint32
-	Stamp           []byte // postage stamp
+	BatchID         []byte  // postage batch ID
+	Sig             []byte  // postage stamp
+	Depth           uint8   // postage batch depth
+	Radius          uint8   // postage batch reserve radius, po upto and excluding which chunks are unpinned
+	Counts          *Counts // postage batch counts
+}
+
+type Counts struct {
+	Counts []byte
+}
+
+func (c *Counts) String() string {
+	counts := make([]uint32, len(c.Counts)/4)
+	for i := range counts {
+		counts[i] = c.add(i, 0, true)
+	}
+	return fmt.Sprintf("%v", counts)
+}
+
+func (c *Counts) Inc(i int) uint32 {
+	return c.add(i, 1, true)
+}
+func (c *Counts) Dec(i int) uint32 {
+	return c.add(i, 1, false)
+}
+
+func (c *Counts) add(i int, d uint32, add bool) uint32 {
+	count := binary.BigEndian.Uint32(c.Counts[4*i : 4*i+4]) // deserialise
```

Comment:
this can panic and take the whole node down

Reply:
how?

Reply:
i don't see where this field is ever initialised. where does c.Counts get allocated? there is no unit test for this, and where are these offsets coming from?